### ISSUE TITLE
Scylla test with all addresses ipv6 public fails on nodetool status

### ISSUE THREAD

**Issue by soyacz** [scylladb/scylla-cluster-tests#7447]
## Issue description

- [ ] This issue is a regression.
- [ ] It is unknown if this issue is a regression.

Tried provision test with `configurations/network_config/all_addresses_ipv6_public.yaml` config and failed on `Waiting for nodes to join the cluster`. 
While nodetool status shows all nodes up:
```
< t:2024-05-17 12:43:13,872 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > Datacenter: eu-west
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > ===================
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > Status=Up/Down
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > |/ State=Normal/Leaving/Joining/Moving
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > -- Address                                 Load    Tokens Owns Host ID                              Rack
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 2a05:d018:12e3:f000:2415:944:1500:6022  1.61 MB 256    ?    a35d9f4b-220a-43d7-8317-bcf2a1e999b3 1a  
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 2a05:d018:12e3:f000:4a6e:f30a:cbf0:f73  1.85 MB 256    ?    dd8fd382-83c8-4e31-a9a7-aaaa3bb98d11 1a  
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 2a05:d018:12e3:f000:4eca:abac:68a4:7d4d 1.67 MB 256    ?    ae2d7ff7-85a6-4b90-9bb4-d39ef1a0d260 1a  
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 2a05:d018:12e3:f000:7a4a:8820:7a5a:24e  1.69 MB 256    ?    63332c60-933b-4f12-bc49-ad7294e7c208 1a  
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 2a05:d018:12e3:f000:ac27:1cec:17ce:814  1.66 MB 256    ?    11ffd883-4b93-49c0-9fd5-410142398859 1a  
< t:2024-05-17 12:43:13,873 f:base.py         l:230  c:RemoteLibSSH2CmdRunner p:DEBUG > UN 2a05:d018:12e3:f000:f763:aad5:adad:144a 1.67 MB 256    ?    6e298767-b691-4915-a693-9a499a15ca99 1a  

```

## Installation details

Cluster size: 6 nodes (i4i.2xlarge)

Scylla Nodes used in this run:
- longevity-10gb-3h-speed-up-db-node-e6323f19-6 (34.246.135.76 | 2a05:d018:12e3:f000:f763:aad5:adad:144a) (shards: 7)
- longevity-10gb-3h-speed-up-db-node-e6323f19-5 (54.247.14.104 | 2a05:d018:12e3:f000:4a6e:f30a:cbf0:f73) (shards: 7)
- longevity-10gb-3h-speed-up-db-node-e6323f19-4 (3.250.174.162 | 2a05:d018:12e3:f000:7a4a:8820:7a5a:24e) (shards: 7)
- longevity-10gb-3h-speed-up-db-node-e6323f19-3 (52.17.71.170 | 2a05:d018:12e3:f000:4eca:abac:68a4:7d4d) (shards: 7)
- longevity-10gb-3h-speed-up-db-node-e6323f19-2 (54.216.72.174 | 2a05:d018:12e3:f000:2415:944:1500:6022) (shards: 7)
- longevity-10gb-3h-speed-up-db-node-e6323f19-1 (3.251.98.1 | 2a05:d018:12e3:f000:ac27:1cec:17ce:814) (shards: 7)

OS / Image: `ami-03e1418be972f5f58` (aws: undefined_region)

Test: `provision-test`
Test id: `e6323f19-83c6-4ebc-97cb-23471b8b64fe`
Test name: `scylla-staging/lukasz/provision-test`
Test config file(s):

- [longevity-10gb-3h.yaml](https://github.com/scylladb/scylla-cluster-tests/blob/6954f412109dd270f350e2bae6d23e64beb192bb/test-cases/longevity/longevity-10gb-3h.yaml)

<details>
<summary>
Logs and commands
</summary>

- Restore Monitor Stack command: `$ hydra investigate show-monitor e6323f19-83c6-4ebc-97cb-23471b8b64fe`
- Restore monitor on AWS instance using [Jenkins job](https://jenkins.scylladb.com/view/QA/job/QA-tools/job/hydra-show-monitor/parambuild/?test_id=e6323f19-83c6-4ebc-97cb-23471b8b64fe)
- Show all stored logs command: `$ hydra investigate show-logs e6323f19-83c6-4ebc-97cb-23471b8b64fe`

## Logs:
- **db-cluster-e6323f19.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/db-cluster-e6323f19.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/db-cluster-e6323f19.tar.gz)
- **sct-runner-events-e6323f19.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/sct-runner-events-e6323f19.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/sct-runner-events-e6323f19.tar.gz)
- **sct-e6323f19.log.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/sct-e6323f19.log.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/sct-e6323f19.log.tar.gz)
- **loader-set-e6323f19.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/loader-set-e6323f19.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/loader-set-e6323f19.tar.gz)
- **monitor-set-e6323f19.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/monitor-set-e6323f19.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/monitor-set-e6323f19.tar.gz)
- **parallel-timelines-report-e6323f19.tar.gz** - [https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/parallel-timelines-report-e6323f19.tar.gz](https://cloudius-jenkins-test.s3.amazonaws.com/e6323f19-83c6-4ebc-97cb-23471b8b64fe/20240517_125013/parallel-timelines-report-e6323f19.tar.gz)

[Jenkins job URL](https://jenkins.scylladb.com/job/scylla-staging/job/lukasz/job/provision-test/22/)
[Argus](https://argus.scylladb.com/test/fb89180f-241e-4341-9c64-bf57d9cbb0c3/runs?additionalRuns[]=e6323f19-83c6-4ebc-97cb-23471b8b64fe)
</details>

**Comment by soyacz** [comment]
also see errors like:
`< t:2024-05-17 12:41:55,817 f:cluster.py      l:2688 c:sdcm.cluster         p:ERROR > Get nodes statuses. Failed to find a node in cluster by IP: 2a05:d018:12e3:f000:ac27:1cec:17ce:0814`

**Comment by fruch** [comment]
Seems like we are comparing again without leading zeros,

I thought I'd fixed it, but it seems we broke it again

**Comment by fruch** [comment]
> Seems like we are comparing again without leading zeros,
> 
> I thought I'd fixed it, but it seems we broke it again

this was the fix: https://github.com/scylladb/scylla-cluster-tests/pull/7047

**Comment by fruch** [comment]
@juliayakovlev please take a look at this one, let's try to find why this isn't working as expected

**Comment by juliayakovlev** [comment]
> > Seems like we are comparing again without leading zeros,
> > I thought I'd fixed it, but it seems we broke it again
> 
> this was the fix: #7047
> 
> and seems like togther with [2c2369d](https://github.com/scylladb/scylla-cluster-tests/commit/2c2369dd228cdef9c326007ebe1f0e2345da43b6) something broke, or AWS started returning the short version of ipv6

I think I need need to add same "magic", like you did in https://github.com/scylladb/scylla-cluster-tests/pull/7047/files, here:

https://github.com/scylladb/scylla-cluster-tests/blob/329157882df843145b0f8522fcd2f2a75fca2b1d/sdcm/cluster_aws.py#L463

**Comment by fruch** [comment]
> > > Seems like we are comparing again without leading zeros,
> > > I thought I'd fixed it, but it seems we broke it again
> > 
> > 
> > this was the fix: #7047
> > and seems like togther with [2c2369d](https://github.com/scylladb/scylla-cluster-tests/commit/2c2369dd228cdef9c326007ebe1f0e2345da43b6) something broke, or AWS started returning the short version of ipv6
> 
> @fruch I think I need need to add similar "magic", like you did in https://github.com/scylladb/scylla-cluster-tests/pull/7047/files, here:
> 
> https://github.com/scylladb/scylla-cluster-tests/blob/329157882df843145b0f8522fcd2f2a75fca2b1d/sdcm/cluster_aws.py#L463

o.k. so lets do (let try doing it on any place we retrieve ipv6 addresses, also for gce/azure ?)

**Comment by juliayakovlev** [comment]
https://github.com/scylladb/scylla-cluster-tests/pull/7460
